tools/c7n_mailer/tests/test_gcp.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2019 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import json
+import unittest
+import zlib
+from common import logger, MAILER_CONFIG_1, GCP_MESSAGE, GCP_MESSAGES
+from c7n_mailer.gcp.gcp_pubsub_processor import MailerGcpPubSubProcessor
+from mock import MagicMock, patch
+
+
+class GcpTest(unittest.TestCase):
+
+    def setUp(self):
+        self.compressed_message = MagicMock()
+        self.compressed_message.content = base64.b64encode(
+            zlib.compress(GCP_MESSAGE.encode('utf8')))
+        self.loaded_message = json.loads(GCP_MESSAGE)
+
+    @patch.object(MailerGcpPubSubProcessor, 'receive_messages')
+    @patch.object(MailerGcpPubSubProcessor, 'process_message')
+    @patch.object(MailerGcpPubSubProcessor, 'ack_messages')
+    def test_process_pubsub_message_success(self, mock_ack, mock_process, mock_receive):
+        mock_process.return_value = True
+        mock_ack.return_value = True
```

Comment:
what is this testing?, your mocking every method of the class except init. none of the actual implementation is being checked.

Reply:
correcting in next commit or two.